pkg/smartcontract/manifest/standard/comply.go

```diff
@@ -73,5 +60,34 @@ func Comply(m, st *manifest.Manifest) error {
 			}
 		}
 	}
+	for _, stm := range st.Optional {
+		if err := checkMethod(m, &stm, true); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func checkMethod(m *manifest.Manifest, expected *manifest.Method, allowMissing bool) error {
+	actual := m.ABI.GetMethod(expected.Name, len(expected.Parameters))
+	if actual == nil {
+		if allowMissing {
+			return nil
+		}
+		return fmt.Errorf("%w: '%s' with %d parameters", ErrMethodMissing,
+			expected.Name, len(expected.Parameters))
+	} else if expected.ReturnType != actual.ReturnType {
+		return fmt.Errorf("%w: '%s' (expected %s, got %s)", ErrInvalidReturnType,
+			expected.Name, expected.ReturnType, actual.ReturnType)
+	}
+	for i := range expected.Parameters {
+		if expected.Parameters[i].Type != actual.Parameters[i].Type {
```

Comment:
Don't we want to check the parameters' names also? It will help us to make our native contracts signatures compatible with C#'s ones and avoid issues like 49b1b4bb27675ef23add9782fd5641c3f3149d3e and 7021c7604205fbf4cbae5bf4a8389021b8e2a073.

Parameters names are strictly defined in the standard.

Reply:
I think we can omit name checks at least for compiler. Names are not currently used by VM and can be pretty annoying (I would never name Go variable `tokenid`, camelCase is prefered).

Reply:
By the way, it is `tokenid` in standard and `tokenId` in C# node.

Reply:
Fixed.